lib/rubocop/cop/performance/detect.rb

```diff
@@ -70,6 +73,12 @@ def preferred_method
           config.for_cop('Style/CollectionMethods') \
             ['PreferredMethods']['detect'] || 'detect'
         end
+
+        def lazy?(node)
+          return false if node.nil?
+          receiver, method, _args = *node
+          method == :lazy && !receiver.nil?
```

Comment:
This logic is a bit flawed, as in classes mixing in Enumerable you can invoke `lazy` with an explicit receiver, but I guess it will do for most people.